skrub/_utils.py

```diff
@@ -156,22 +159,85 @@ def repr_args(args, kwargs, defaults={}):
     )
 
 
-def transformer_output_type_error(transformer, transform_input, transform_output):
-    module = sbd.dataframe_module_name(transform_input)
+def set_output(transformer, X):
+    if not hasattr(transformer, "set_output"):
+        return
+    module_name = sbd.dataframe_module_name(X)
+    if module_name == "polars" and parse_version(sklearn.__version__) < parse_version(
+        "1.4"
+    ):
+        # TODO: remove when scikit-learn 1.3 support is dropped.
+        warnings.warn(
```

Comment:
I'm not sure if we should have this warning. Hopefully in a growing number of cases polars can do the conversion from pandas without a copy

Reply:
I would skip the warning. At least with a have a better support in new version but warning could be quite annoying.